index.js

```diff
@@ -1,40 +1,89 @@
-const express = require("express");
+const fastcsv = require('fast-csv');
+const fs = require('fs');
+const ws = fs.createWriteStream('livingstonProperties.csv');
+const express = require('express');
+const puppeteer = require('puppeteer');
 const app = express();
 const port = 6666;
-
-const puppeteer = require("puppeteer");
-
 const parseAddresses = require('./addressParser');
 
-const url = "https://property.franklincountyauditor.com/_web/search/commonsearch.aspx?mode=address";
+let resultsArray = [];
+
+const url =
+  'https://property.franklincountyauditor.com/_web/search/commonsearch.aspx?mode=address';
 const addresses = parseAddresses();
 
-async function configureTheBrowser(){
-	// const browswer = await puppeteer.launch();
-	const browswer = await puppeteer.launch({ headless: false});
-	const page = await browswer.newPage();
-	await page.goto(url, { waitUntil: "load", timeout: 0});
-	return page;
+async function searchForAddress(page, streetNumber) {
+  if (streetNumber) {
+    await page.waitForSelector('#inpNumber');
+    await page.type('#inpNumber', String(streetNumber));
+    await page.waitForSelector('#inpStreet');
+    await page.type('#inpStreet', 'Livingston');
+    await Promise.all([
+      page.waitForNavigation(),
+      page.$eval('button[name=btSearch]', (element) => element.click()),
+    ]);
+
+    await scrapeData(page);
+
+    await page.close();
+  }
 }
 
-async function searchForAddress(page, streetNumber){
-	console.log("Searching for address...")
-	if(streetNumber){
-		console.log(String(streetNumber))
-		await page.waitForSelector('input[name=inpNumber]');
-		await page.$eval('input[name=inpNumber]',element => element.value = String(streetNumber))
-		await page.$eval('input[name=inpStreet]', element => element.value = "Livingston" )
-		await page.$eval('button[name=btSearch]', element => element.click())
-	}
+async function scrapeData(page) {
+  // Find all the elements with a className of 'DataletSideHeading'...
+  // ... loop through to add the key/value pair into the resultsArray
+  const resultObj = await page.$$eval('.DataletSideHeading', (titles) => {
```

Comment:
Just curious about the double `$` for the `$$eval` handler. I've seen `$eval` but not `$$eval`. What's the difference here?

Reply:
In my understanding, $eval selects a single html element or the first element with the same identifier. While $$eval selects multiple html elements.